packages/utils/src/libp2p/dialHelper.ts

```diff
@@ -123,11 +131,17 @@ async function queryDHT(
   libp2p: ReducedDHT,
   destination: PeerId,
   opts: Required<TimeoutOpts>
-): Promise<DialResponse | { status: InternalDialStatus.CONTINUE; dhtResponse: DHTResponse }> {
-  let dhtResponse: DHTResponse
+): Promise<DialResponse | { status: InternalDialStatus.CONTINUE; relayers: Multiaddr[] }> {
+  const relayers: Relayers[] = []
+
+  const key = await createRelayerKey(destination)
+  log(`fetching relay keys for node ${destination.toB58String()} from DHT.`, key)
   try {
-    // Let libp2p populate its internal peerStore with fresh addresses
-    dhtResponse = await libp2p.peerRouting.findPeer(destination, { timeout: DEFAULT_DHT_QUERY_TIMEOUT })
+    for await (const relayer of libp2p.contentRouting.findProviders(key, {
```

Comment:
Previously:
Use DHT to "look" for latest signed peer record of destination
Now:
Ask the DHT "Who has a relayer token for the destination?"